test/spec/proxySpec.js

```diff
@@ -1148,6 +1146,86 @@ describe('proxy', function() {
       // }, 100);
     });
   });
+  describe('Already proxified values', function() {
+    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
```

Comment:
I do acknowledge that it is indeed potentially confusing to developers, and I can overcome that limitation since now we have a cache of all `this` instance's proxies. I can tell whether a proxified object is alien and just re-proxify it. That probably works since proxification is 100% transparent. 

But IMO, I'd rather have:
```js 
ProxyB.proxifiedObject.add // undefined
``` 
Than have two proxies emitting patches together. And it's not a matter of taste, just being consistent. Becuase proxifying an object deep clones it, why doing that in runtime should be different?

You might ask why do we deep clone objects before proxification on the first place? Because if not,  you'll be able to sneak non-proxified objects into the proxified tree causing a lot of nightmarish bugs.